csharp/src/Apache.Arrow/ArrowBuffer.Builder.cs

```diff
@@ -113,29 +205,28 @@ public ArrowBuffer Build(MemoryAllocator allocator = default)
                 return new ArrowBuffer(memoryOwner);
             }
 
-            private void EnsureCapacity(int n)
+            private void EnsureAdditionalCapacity(int additionalCapacity)
             {
-                var length = checked(Length + n);
+                EnsureCapacity(checked(Length + additionalCapacity));
+            }
 
-                if (length > Capacity)
+            private void EnsureCapacity(int requiredCapacity)
+            {
+                if (requiredCapacity > Capacity)
                 {
                     // TODO: specifiable growth strategy
-
-                    var capacity = Math.Max(length * _size, Memory.Length * 2);
+                    // Double the length of the in-memory array, or use the byte count of the capacity, whichever is
+                    // greater.
+                    var capacity = Math.Max(requiredCapacity * _size, Memory.Length * 2);
                     Reallocate(capacity);
                 }
             }
 
-            private void Reallocate(int length)
+            private void Reallocate(int numBytes)
```

Comment:
The variable name `length` was previously used both to mean number of items (i.e. `ArrowBuffer.Builder.Length`) but also here to mean number of bytes.  Renamed the local variable to remove any ambiguity.